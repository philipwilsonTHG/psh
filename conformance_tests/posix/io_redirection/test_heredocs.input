# Test here documents

# Basic here document
cat << EOF
This is a basic here document
with multiple lines
of text.
EOF

# Here document with variable expansion
name="PSH"
cat << EOF
Hello from $name
The shell name is: $name
EOF

# Here document with indented terminator (<<-)
cat <<- EOF
	This is indented text
	The leading tabs should be stripped
	from both the content and terminator
	EOF

# Here document with quoted delimiter (no expansion)
cat << 'EOF'
This should not expand: $name
Literal dollar signs: $HOME
EOF

# Here document redirected to file
mkdir -p tmp
cat << EOF > tmp/heredoc_test.txt
This goes to a file
Line 2 of the file
EOF
cat tmp/heredoc_test.txt

# Cleanup
rm -rf tmp