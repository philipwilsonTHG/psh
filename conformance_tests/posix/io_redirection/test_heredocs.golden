This is a basic here document
with multiple lines
of text.
Hello from PSH
The shell name is: PSH
This is indented text
The leading tabs should be stripped
from both the content and terminator
This should not expand: $name
Literal dollar signs: $HOME
This goes to a file
Line 2 of the file
